honeybadgermpc/mimc.py

```diff
@@ -0,0 +1,64 @@
+from math import log, ceil
+import logging
+import asyncio
+from honeybadgermpc.mpc import PreProcessedElements, TaskProgramRunner, Subgroup
+from honeybadgermpc.field import GF
+from honeybadgermpc.mixins import BeaverTriple, MixinOpName
+
+# ROUND: iteration time of MiMC encryption function, r = ceil(log(p, 3)) = 161
+ROUND = ceil(log(Subgroup.BLS12_381))
+
+
+async def mimc(context, x, k):
+    # message x \in Fp, secret key k \in Fp
+
+    pp_elements = PreProcessedElements()
+
+    # def cube(): [x] -> [x^3]
+    async def cube(x):
+        # first compute a tuple ([r], [r^2], [r^3])
+        r = pp_elements.get_rand(context)
+        r2 = await (r * r)
+        r3 = await (r * r2)
```

Comment:
We should ideally be getting the entire [r] [r^2] [r^3] from preprocessing, similar to powermix

Reply:
Made it as a separate PR.